### PR TITLE
Update the list of confusable characters

### LINKED ISSUE #43629: unicode_chars.rs should be updated to match the latest confusables document
`src/libsyntax/parse/lexer/unicode_chars.rs` contains a list of confusable characters (see #25957). Its comments mention that it bases on a [document](http://www.unicode.org/Public/security/revision-06/confusables.txt). Of this document, a newer revision has been published ([link](http://www.unicode.org/Public/security/10.0.0/confusables.txt)). It contains many more confusables, e.g. ᐸ (canadian syllabics Pa). We should update unicode_chars.rs to include the newly listed confusables as well.

Bug report inspired by [this reddit post](https://www.reddit.com/r/rustjerk/comments/6c7ym4/when_the_compiler_has_your_back/disnn7g/).

### PR DESCRIPTION
**Pull request by MaloJaffre**
Also reorder and space the list to make it clearer for futures updates
and to come closer to the original list.

This was tedious but somewhat rewarding!

Thanks @est31 for the instructions.

Fixes #43629.
r? @est31

### CODE REVIEW

**Review by est31 — APPROVED** [review]
Thanks!

File: src/libsyntax/parse/lexer/unicode_chars.rs [hunk 1]
```diff
@@ -9,55 +9,80 @@
...
+    (' ', "No-Break Space", ' '),
+    (' ', "Figure Space", ' '),
+    (' ', "Narrow No-Break Space", ' '),
     ('　', "Ideographic Space", ' '),
+
     ('ߺ', "Nko Lajanyalan", '_'),
     ('﹍', "Dashed Low Line", '_'),
     ('﹎', "Centreline Low Line", '_'),
     ('﹏', "Wavy Low Line", '_'),
+    ('＿', "Fullwidth Low Line", '-'),
```
Comment (ollie27) [inline src/libsyntax/parse/lexer/unicode_chars.rs]: Shouldn't this be `'_'`?

Comment (MaloJaffre) [inline src/libsyntax/parse/lexer/unicode_chars.rs]: Good catch.
Thanks!

**Comment by ubsan** [comment]
`runic multiple ponctuation`

**Comment by GuillaumeGomez** [comment]
Looks good to me but I'm not sure I should the one r+ing it.

r? @eddyb

**Comment by ubsan** [comment]
Wait, what? that was pointing out a typo.

**Comment by ubsan** [comment]
Why was this merged with the typo -.-

**Comment by MaloJaffre** [comment]
My bad, I was thinking it was a joke and didn't saw the typo.

**Comment by ubsan** [comment]
@MaloJaffre you're fine lol. I just got my panties in a twist.

**Pull request merged**